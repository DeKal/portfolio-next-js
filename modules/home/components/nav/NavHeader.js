import React from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import { Translate } from 'react-localize-redux'
import { GiHamburgerMenu } from 'react-icons/gi'
import { MdClose } from 'react-icons/md'
import NavLang from '~/modules/home/components/nav/NavLang'
import MobileContainer from '~/modules/home/components/common/container/MobileContainer'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'

const NavHeader = ({ isShowNavContent, switchNav }) => (
  <NavContainer>
    <MobileContainer>
      <Row fluid="true">
        <Col xs={6}>
          <NavLang />
        </Col>
        <Col xs={6}>
          <NavControl
            data-test-id="switch-nav-content"
            onClick={() => switchNav(!isShowNavContent)}
          >
            {!isShowNavContent ? <HamburgerIcon /> : <CloseIcon />}{' '}
            <Translate id="Menu" />
          </NavControl>
        </Col>
      </Row>
    </MobileContainer>
  </NavContainer>
)

NavHeader.propTypes = {
  isShowNavContent: PropTypes.bool,
  switchNav: PropTypes.func
}

export default NavHeader

const NavContainer = styled.div`
  width: 100%;
  position: absolute;
  z-index: 200;
  top: 0;
  left: 0;
  height: ${props => props.theme.navHeaderHeight};
  overflow: hidden;
  background-color: ${props => props.theme.colors.colorAccent};
  box-shadow: ${props => props.theme.extra.boxShadow};
`

const NavControl = styled.div`
  color: ${props => props.theme.colors.colorDark};
  line-height: ${props => props.theme.navHeaderHeight};
  text-align: right;
  border: none;
  transition: color 0.5s;
  cursor: pointer;

  @media screen and (max-width: 768px) {
    visibility: hidden;
  }

  &::before {
    display: inline-block;
    margin-right: 10px;
    color: ${props => props.theme.colors.colorDark};
    font-size: ${props => props.theme.fontSizes.XL};
    transition: color 0.5s;

    @media screen and (max-width: 600px) {
      margin-right: 0;
    }
  }

  &:hover,
  &:hover::before {
    color: ${props => props.theme.colors.colorDark};
  }
`

const HamburgerIcon = styled(GiHamburgerMenu)`
  display: inline;
  padding-bottom: 5px;
`

const CloseIcon = styled(MdClose)`
  display: inline;
  font-size: ${props => props.theme.fontSizes.XXL};
  font-weight: bolder;
  padding-bottom: 3px;
`
