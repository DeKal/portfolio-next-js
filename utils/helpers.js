import { INIT_SELECTED_PAGE } from '~/consts/initState'
import $ from 'jquery'

export const initHashLocation = () => {
  if (location.hash) {
    const offset = $(location.hash).offset()
    window.scrollTo(offset.left, offset.top)
  }
}

export const isSafari = () => {
  var userAgent = navigator.userAgent.toLowerCase()
  if (userAgent.indexOf('safari') != -1) {
    // Safari agent will not have Chrome
    if (userAgent.indexOf('chrome') > -1) {
      return false
    } else {
      return true
    }
  }
}

export const setJqueryScrollEvent = () => {
  $('a').on('click', function(event) {
    // Make sure this.hash has a value before overriding default behavior
    if (this.hash) {
      // Prevent default anchor click behavior
      event.preventDefault()
      // Store hash
      var hash = this.hash

      $('html, body').animate(
        {
          scrollTop: $(hash).offset().top
        },
        400,
        function() {
          // Add hash (#) to URL when done scrolling (default click behavior)
          window.location.hash = hash
        }
      )
    } // End if
  })
}

export const cleanUpScrollEvent = () => {
  $('a').unbind()
}

export const getSelectedSection = () => {
  if (location.hash.length !== 0) {
    const selectedPage = location.hash.substr(1)
    return selectedPage
  }
  return INIT_SELECTED_PAGE
}

export const isChristmasDay = () => {
  const today = new Date()
  const month = today.getMonth() + 1
  const isDecember = month == 12
  const day = today.getDate()
  const isChristMasTime = 20 <= day <= 30

  return isDecember && isChristMasTime
}

export const isNewYearComing = () => {
  const today = new Date()
  const day = today.getDate()
  const isFirstTwoWeek = 1 <= day <= 16
  const month = today.getMonth() + 1
  const isJanuary = month == 1

  return isJanuary && isFirstTwoWeek
}
