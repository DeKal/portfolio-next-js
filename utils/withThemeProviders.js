import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import defaultTheme from '~/views/theme'

export const shallowWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

export const mountWithTheme = (children, theme = defaultTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}
