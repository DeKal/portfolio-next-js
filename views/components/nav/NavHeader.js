import React from 'react'
import styled from 'styled-components/macro'
import { GiHamburgerMenu } from 'react-icons/gi'
import { MdClose } from 'react-icons/md'
import NavLang from '~/views/components/nav/NavLang'
import PropTypes from 'prop-types'

const NavHeader = ({ isShowNavContent, switchNav }) => (
  <NavContainer className="first-nav">
    <div className="container">
      <NavLang />
      <NavControl
        data-test-id="switch-nav-content"
        className="first-nav__navcontrol"
        onClick={() => switchNav(!isShowNavContent)}
        href="#"
      >
        {!isShowNavContent ? <HamburgerIcon /> : <CloseIcon />} Menu
      </NavControl>
    </div>
  </NavContainer>
)

NavHeader.propTypes = {
  isShowNavContent: PropTypes.bool,
  switchNav: PropTypes.func
}

export default NavHeader

const NavContainer = styled.div`
  position: absolute;
  z-index: 200;
  top: 0;
  left: 0;
  height: ${props => props.theme.navHeaderHeight};
  overflow: hidden;
  background-color: ${props => props.theme.colors.colorAccent};
  box-shadow: ${props => props.theme.extra.boxShadow};
`

const NavControl = styled.a`
  color: ${props => props.theme.colors.colorLight};
  line-height: ${props => props.theme.navHeaderHeight};
  text-align: right;
  border: none;
  transition: color 0.5s;

  @media screen and (max-width: 600px) {
    margin-left: 0;
    font-size: 0;
  }

  &::before {
    display: inline-block;
    margin-right: 10px;
    color: ${props => props.theme.colors.colorLight};
    font-size: ${props => props.theme.fontSizes.XL};
    transition: color 0.5s;

    @media screen and (max-width: 600px) {
      margin-right: 0;
    }
  }

  &.is-open,
  &.is-closed {
    display: block;
  }

  &:hover,
  &:hover::before {
    color: ${props => props.theme.colors.colorDark};
  }
`

const HamburgerIcon = styled(GiHamburgerMenu)`
  display: inline;
  padding-bottom: 5px;
`

const CloseIcon = styled(MdClose)`
  display: inline;
  font-size: ${props => props.theme.fontSizes.XXL};
  font-weight: bolder;
  padding-bottom: 3px;
`
