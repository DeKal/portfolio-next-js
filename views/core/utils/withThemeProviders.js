import { shallow, mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components/macro'
import homeTheme from '~/theme/theme'
import baseTheme from '~/theme/baseTheme'

/* istanbul ignore next */
export const shallowWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}

/* istanbul ignore next */
export const shallowWithDivingTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children).dive()
}

/* istanbul ignore next */
export const mountWithTheme = (children, theme = homeTheme) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}

/* istanbul ignore next */
export const shallowWithBaseTheme = (children, theme = baseTheme) => {
  ThemeConsumer._currentValue = theme
  return shallow(children)
}
