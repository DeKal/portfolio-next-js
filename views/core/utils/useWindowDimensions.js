import { useState, useEffect } from 'react'
import { getWindowDimensions } from '~/views/core/utils/helpers'

/* istanbul ignore next */
const useWindowDimensions = () => {
  const [windowDimensions, setWindowDimensions] = useState(
    getWindowDimensions()
  )

  useEffect(() => {
    /* istanbul ignore next */
    function handleResize() {
      setWindowDimensions(getWindowDimensions())
    }

    window.addEventListener('resize', handleResize)
    /* istanbul ignore next */
    return () => window.removeEventListener('resize', handleResize)
  }, [])

  return windowDimensions
}

export default useWindowDimensions
