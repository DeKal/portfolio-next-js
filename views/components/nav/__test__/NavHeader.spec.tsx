import * as React from 'react'
import { mountWithTheme, shallowWithTheme } from '~/utils/withThemeProviders'
import NavHeader from '~/views/components/nav/NavHeader'

describe('Nav Header', () => {
  it('should render without throwing an error', function() {
    const wrap = mountWithTheme(<NavHeader />)

    expect(shallowWithTheme(<NavHeader />)).toMatchSnapshot()
  })

  it('should switchNav from Off to On', function() {
    const switchNav = jest.fn()
    const wrap = shallowWithTheme(
      <NavHeader isShowNavContent={false} switchNav={switchNav} />
    )
    expect(shallowWithTheme(<NavHeader />)).toMatchSnapshot()

    wrap.find('[data-test-id="switch-nav-content"]').simulate('click')
    
    expect(switchNav.mock.calls.length).toEqual(1)
    expect(switchNav).toHaveBeenCalledWith(true);
  })

  it('should switchNav from On to Off', function() {
    const switchNav = jest.fn()
    const wrap = shallowWithTheme(
      <NavHeader isShowNavContent switchNav={switchNav} />
    )
    expect(shallowWithTheme(<NavHeader />)).toMatchSnapshot()

    wrap.find('[data-test-id="switch-nav-content"]').simulate('click')
    
    expect(switchNav.mock.calls.length).toEqual(1)
    expect(switchNav).toHaveBeenCalledWith(false);
    
  })
})
