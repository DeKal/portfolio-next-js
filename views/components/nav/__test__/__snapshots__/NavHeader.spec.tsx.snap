// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Header should render when not show NavContent 1`] = `
<NavHeader>
  <NavHeader__NavContainer>
    <div
      className="NavHeader__NavContainer-twnsn2-0 hBaJqf"
    >
      <MobileContainer>
        <Container
          className="MobileContainer-sc-14bpgy4-0 cmmyDP"
          fluid={false}
        >
          <div
            className="MobileContainer-sc-14bpgy4-0 cmmyDP container"
          >
            <ForwardRef
              fluid="true"
              noGutters={false}
            >
              <div
                className="row"
                fluid="true"
              >
                <Col
                  xs={6}
                >
                  <div
                    className="col-6"
                  >
                    <NavLang>
                      <NavLang__NotSelectableDiv>
                        <nav
                          className="NavLang__NotSelectableDiv-sc-7efc3x-0 gkLMmc"
                        >
                          <ul>
                            <NavLang__NavLinkItem>
                              <li
                                className="NavLang__NavLinkItem-sc-7efc3x-1 iyUXK"
                              >
                                <LocalizedComponent
                                  lang="en"
                                  selectedLang="en"
                                  setSelectedLang={[Function]}
                                >
                                  <NavLink
                                    addTranslation={[Function]}
                                    addTranslationForLanguage={[Function]}
                                    ignoreTranslateChildren={false}
                                    initialize={[Function]}
                                    lang="en"
                                    languages={Array []}
                                    renderToStaticMarkup={false}
                                    selectedLang="en"
                                    setActiveLanguage={[Function]}
                                    setSelectedLang={[Function]}
                                    translate={[Function]}
                                  >
                                    <NavLink__Container
                                      isSelected={true}
                                      onClick={[Function]}
                                    >
                                      <a
                                        className="NavLink__Container-v6tuz9-0 dUOeKU"
                                        onClick={[Function]}
                                      >
                                        <Translate
                                          id="english"
                                        >
                                          <WrappedTranslate
                                            context={
                                              Object {
                                                "activeLanguage": undefined,
                                                "addTranslation": [Function],
                                                "addTranslationForLanguage": [Function],
                                                "defaultLanguage": undefined,
                                                "ignoreTranslateChildren": false,
                                                "initialize": [Function],
                                                "languages": Array [],
                                                "renderToStaticMarkup": false,
                                                "setActiveLanguage": [Function],
                                                "translate": [Function],
                                              }
                                            }
                                            id="english"
                                          >
                                            Missing translationId: english for language: \${ languageCode }
                                          </WrappedTranslate>
                                        </Translate>
                                      </a>
                                    </NavLink__Container>
                                  </NavLink>
                                </LocalizedComponent>
                                <NavLang__InlineBlock>
                                  <div
                                    className="NavLang__InlineBlock-sc-7efc3x-2 gQsqjl"
                                  >
                                    /
                                  </div>
                                </NavLang__InlineBlock>
                                <LocalizedComponent
                                  lang="vn"
                                  selectedLang="en"
                                  setSelectedLang={[Function]}
                                >
                                  <NavLink
                                    addTranslation={[Function]}
                                    addTranslationForLanguage={[Function]}
                                    ignoreTranslateChildren={false}
                                    initialize={[Function]}
                                    lang="vn"
                                    languages={Array []}
                                    renderToStaticMarkup={false}
                                    selectedLang="en"
                                    setActiveLanguage={[Function]}
                                    setSelectedLang={[Function]}
                                    translate={[Function]}
                                  >
                                    <NavLink__Container
                                      isSelected={false}
                                      onClick={[Function]}
                                    >
                                      <a
                                        className="NavLink__Container-v6tuz9-0 ddSDsj"
                                        onClick={[Function]}
                                      >
                                        <Translate
                                          id="vietnamese"
                                        >
                                          <WrappedTranslate
                                            context={
                                              Object {
                                                "activeLanguage": undefined,
                                                "addTranslation": [Function],
                                                "addTranslationForLanguage": [Function],
                                                "defaultLanguage": undefined,
                                                "ignoreTranslateChildren": false,
                                                "initialize": [Function],
                                                "languages": Array [],
                                                "renderToStaticMarkup": false,
                                                "setActiveLanguage": [Function],
                                                "translate": [Function],
                                              }
                                            }
                                            id="vietnamese"
                                          >
                                            Missing translationId: vietnamese for language: \${ languageCode }
                                          </WrappedTranslate>
                                        </Translate>
                                      </a>
                                    </NavLink__Container>
                                  </NavLink>
                                </LocalizedComponent>
                              </li>
                            </NavLang__NavLinkItem>
                          </ul>
                        </nav>
                      </NavLang__NotSelectableDiv>
                    </NavLang>
                  </div>
                </Col>
                <Col
                  xs={6}
                >
                  <div
                    className="col-6"
                  >
                    <NavHeader__NavControl
                      data-test-id="switch-nav-content"
                      onClick={[Function]}
                    >
                      <div
                        className="NavHeader__NavControl-twnsn2-1 grKlkF"
                        data-test-id="switch-nav-content"
                        onClick={[Function]}
                      >
                        <NavHeader__HamburgerIcon>
                          <GiHamburgerMenu
                            className="NavHeader__HamburgerIcon-twnsn2-2 cJgLJz"
                          >
                            <IconBase
                              attr={
                                Object {
                                  "viewBox": "0 0 512 512",
                                }
                              }
                              className="NavHeader__HamburgerIcon-twnsn2-2 cJgLJz"
                            >
                              <svg
                                className="NavHeader__HamburgerIcon-twnsn2-2 cJgLJz"
                                fill="currentColor"
                                height="1em"
                                stroke="currentColor"
                                strokeWidth="0"
                                style={
                                  Object {
                                    "color": undefined,
                                  }
                                }
                                viewBox="0 0 512 512"
                                width="1em"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
                                  key="0"
                                />
                              </svg>
                            </IconBase>
                          </GiHamburgerMenu>
                        </NavHeader__HamburgerIcon>
                         
                        <Translate
                          id="Menu"
                        >
                          <WrappedTranslate
                            context={
                              Object {
                                "activeLanguage": undefined,
                                "addTranslation": [Function],
                                "addTranslationForLanguage": [Function],
                                "defaultLanguage": undefined,
                                "ignoreTranslateChildren": false,
                                "initialize": [Function],
                                "languages": Array [],
                                "renderToStaticMarkup": false,
                                "setActiveLanguage": [Function],
                                "translate": [Function],
                              }
                            }
                            id="Menu"
                          >
                            Missing translationId: Menu for language: \${ languageCode }
                          </WrappedTranslate>
                        </Translate>
                      </div>
                    </NavHeader__NavControl>
                  </div>
                </Col>
              </div>
            </ForwardRef>
          </div>
        </Container>
      </MobileContainer>
    </div>
  </NavHeader__NavContainer>
</NavHeader>
`;

exports[`Nav Header should render when show NavContent 1`] = `
<NavHeader
  isShowNavContent={true}
>
  <NavHeader__NavContainer>
    <div
      className="NavHeader__NavContainer-twnsn2-0 hBaJqf"
    >
      <MobileContainer>
        <Container
          className="MobileContainer-sc-14bpgy4-0 cmmyDP"
          fluid={false}
        >
          <div
            className="MobileContainer-sc-14bpgy4-0 cmmyDP container"
          >
            <ForwardRef
              fluid="true"
              noGutters={false}
            >
              <div
                className="row"
                fluid="true"
              >
                <Col
                  xs={6}
                >
                  <div
                    className="col-6"
                  >
                    <NavLang>
                      <NavLang__NotSelectableDiv>
                        <nav
                          className="NavLang__NotSelectableDiv-sc-7efc3x-0 gkLMmc"
                        >
                          <ul>
                            <NavLang__NavLinkItem>
                              <li
                                className="NavLang__NavLinkItem-sc-7efc3x-1 iyUXK"
                              >
                                <LocalizedComponent
                                  lang="en"
                                  selectedLang="en"
                                  setSelectedLang={[Function]}
                                >
                                  <NavLink
                                    addTranslation={[Function]}
                                    addTranslationForLanguage={[Function]}
                                    ignoreTranslateChildren={false}
                                    initialize={[Function]}
                                    lang="en"
                                    languages={Array []}
                                    renderToStaticMarkup={false}
                                    selectedLang="en"
                                    setActiveLanguage={[Function]}
                                    setSelectedLang={[Function]}
                                    translate={[Function]}
                                  >
                                    <NavLink__Container
                                      isSelected={true}
                                      onClick={[Function]}
                                    >
                                      <a
                                        className="NavLink__Container-v6tuz9-0 dUOeKU"
                                        onClick={[Function]}
                                      >
                                        <Translate
                                          id="english"
                                        >
                                          <WrappedTranslate
                                            context={
                                              Object {
                                                "activeLanguage": undefined,
                                                "addTranslation": [Function],
                                                "addTranslationForLanguage": [Function],
                                                "defaultLanguage": undefined,
                                                "ignoreTranslateChildren": false,
                                                "initialize": [Function],
                                                "languages": Array [],
                                                "renderToStaticMarkup": false,
                                                "setActiveLanguage": [Function],
                                                "translate": [Function],
                                              }
                                            }
                                            id="english"
                                          >
                                            Missing translationId: english for language: \${ languageCode }
                                          </WrappedTranslate>
                                        </Translate>
                                      </a>
                                    </NavLink__Container>
                                  </NavLink>
                                </LocalizedComponent>
                                <NavLang__InlineBlock>
                                  <div
                                    className="NavLang__InlineBlock-sc-7efc3x-2 gQsqjl"
                                  >
                                    /
                                  </div>
                                </NavLang__InlineBlock>
                                <LocalizedComponent
                                  lang="vn"
                                  selectedLang="en"
                                  setSelectedLang={[Function]}
                                >
                                  <NavLink
                                    addTranslation={[Function]}
                                    addTranslationForLanguage={[Function]}
                                    ignoreTranslateChildren={false}
                                    initialize={[Function]}
                                    lang="vn"
                                    languages={Array []}
                                    renderToStaticMarkup={false}
                                    selectedLang="en"
                                    setActiveLanguage={[Function]}
                                    setSelectedLang={[Function]}
                                    translate={[Function]}
                                  >
                                    <NavLink__Container
                                      isSelected={false}
                                      onClick={[Function]}
                                    >
                                      <a
                                        className="NavLink__Container-v6tuz9-0 ddSDsj"
                                        onClick={[Function]}
                                      >
                                        <Translate
                                          id="vietnamese"
                                        >
                                          <WrappedTranslate
                                            context={
                                              Object {
                                                "activeLanguage": undefined,
                                                "addTranslation": [Function],
                                                "addTranslationForLanguage": [Function],
                                                "defaultLanguage": undefined,
                                                "ignoreTranslateChildren": false,
                                                "initialize": [Function],
                                                "languages": Array [],
                                                "renderToStaticMarkup": false,
                                                "setActiveLanguage": [Function],
                                                "translate": [Function],
                                              }
                                            }
                                            id="vietnamese"
                                          >
                                            Missing translationId: vietnamese for language: \${ languageCode }
                                          </WrappedTranslate>
                                        </Translate>
                                      </a>
                                    </NavLink__Container>
                                  </NavLink>
                                </LocalizedComponent>
                              </li>
                            </NavLang__NavLinkItem>
                          </ul>
                        </nav>
                      </NavLang__NotSelectableDiv>
                    </NavLang>
                  </div>
                </Col>
                <Col
                  xs={6}
                >
                  <div
                    className="col-6"
                  >
                    <NavHeader__NavControl
                      data-test-id="switch-nav-content"
                      onClick={[Function]}
                    >
                      <div
                        className="NavHeader__NavControl-twnsn2-1 grKlkF"
                        data-test-id="switch-nav-content"
                        onClick={[Function]}
                      >
                        <NavHeader__CloseIcon>
                          <MdClose
                            className="NavHeader__CloseIcon-twnsn2-3 hqrtn"
                          >
                            <IconBase
                              attr={
                                Object {
                                  "viewBox": "0 0 24 24",
                                }
                              }
                              className="NavHeader__CloseIcon-twnsn2-3 hqrtn"
                            >
                              <svg
                                className="NavHeader__CloseIcon-twnsn2-3 hqrtn"
                                fill="currentColor"
                                height="1em"
                                stroke="currentColor"
                                strokeWidth="0"
                                style={
                                  Object {
                                    "color": undefined,
                                  }
                                }
                                viewBox="0 0 24 24"
                                width="1em"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                                  key="0"
                                />
                              </svg>
                            </IconBase>
                          </MdClose>
                        </NavHeader__CloseIcon>
                         
                        <Translate
                          id="Menu"
                        >
                          <WrappedTranslate
                            context={
                              Object {
                                "activeLanguage": undefined,
                                "addTranslation": [Function],
                                "addTranslationForLanguage": [Function],
                                "defaultLanguage": undefined,
                                "ignoreTranslateChildren": false,
                                "initialize": [Function],
                                "languages": Array [],
                                "renderToStaticMarkup": false,
                                "setActiveLanguage": [Function],
                                "translate": [Function],
                              }
                            }
                            id="Menu"
                          >
                            Missing translationId: Menu for language: \${ languageCode }
                          </WrappedTranslate>
                        </Translate>
                      </div>
                    </NavHeader__NavControl>
                  </div>
                </Col>
              </div>
            </ForwardRef>
          </div>
        </Container>
      </MobileContainer>
    </div>
  </NavHeader__NavContainer>
</NavHeader>
`;

exports[`Nav Header should switchNav from Off to On 1`] = `
<NavHeader__NavContainer>
  <MobileContainer>
    <ForwardRef
      fluid="true"
      noGutters={false}
    >
      <Col
        xs={6}
      >
        <NavLang />
      </Col>
      <Col
        xs={6}
      >
        <NavHeader__NavControl
          data-test-id="switch-nav-content"
          onClick={[Function]}
        >
          <NavHeader__HamburgerIcon />
           
          <Translate
            id="Menu"
          />
        </NavHeader__NavControl>
      </Col>
    </ForwardRef>
  </MobileContainer>
</NavHeader__NavContainer>
`;

exports[`Nav Header should switchNav from On to Off 1`] = `
<NavHeader__NavContainer>
  <MobileContainer>
    <ForwardRef
      fluid="true"
      noGutters={false}
    >
      <Col
        xs={6}
      >
        <NavLang />
      </Col>
      <Col
        xs={6}
      >
        <NavHeader__NavControl
          data-test-id="switch-nav-content"
          onClick={[Function]}
        >
          <NavHeader__CloseIcon />
           
          <Translate
            id="Menu"
          />
        </NavHeader__NavControl>
      </Col>
    </ForwardRef>
  </MobileContainer>
</NavHeader__NavContainer>
`;
